kong/tools/request_aware_table.lua

```diff
@@ -0,0 +1,72 @@
+--- NOTE: tool is designed to assist with **detecting** request contamination
+-- issues on CI, during test runs. It does not offer security safeguards.
+
+local table_new = require "table.new"
+local base = require "resty.core.base"
+
+local debug_mode  = kong.configuration.log_level == "debug"
+local get_request = base.get_request
+local var         = ngx.var
+
+
+-- Check if access is allowed for table, based on the request ID
+local function enforce_sequential_access(table)
+  if next(table.__data) == nil or not get_request() then
```

Comment:
Do we need the empty `__data` table check? I thought just checking `__allowed_request_id` below is enough.

Reply:
the empty `__data` check is used to reset `__allowed_request_id` when the table is cleared. If there is no data it means the table was emptied. This is used to avoid having to specify a `:clear` method.

Reply:
@samugi I see. `:clear` would be a safer approach and I think we should keep that. It also makes the clearing intention more explicit.